community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml

```diff
@@ -20,55 +20,53 @@
     job_queue_ha: false
     spool_dir: /var/lib/condor/spool
     condor_config_root: /etc/condor
-    role_file: 00-role
-    pool_file: 01-pool
-    cm_config_file: 02-central-manager
-    cm_ha_config_file: 02-central-manager-high-availability
-    schedd_config_file: 02-schedd
-    schedd_ha_config_file: 02-schedd-high-availability
+    ghpc_config_file: 50-ghpc-managed
+    schedd_ha_config_file: 51-ghpc-schedd-high-availability
     execute_config_file: 02-execute
   tasks:
-  - name: User must supply HTCondor role
+  - name: Ensure necessary variables are set
     ansible.builtin.assert:
       that:
-      - htcondor_central_manager_ips is defined
       - htcondor_role is defined
       - password_id is defined
-      - project_id is defined
-  - name: Set Trust Domain
-    ansible.builtin.set_fact:
-      trust_domain: c.{{ project_id }}.internal
+      - trust_domain is defined
+      - config_object is defined
   - name: Set HTCondor Pool password (token signing key)
     ansible.builtin.shell: |
       set -e -o pipefail
       POOL_PASSWORD=$(gcloud secrets versions access latest --secret={{ password_id }})
       echo -n "$POOL_PASSWORD" | sh -c "condor_store_cred add -c -i -"
     args:
       creates: "{{ condor_config_root }}/passwords.d/POOL"
+      executable: /bin/bash
   - name: Remove default HTCondor configuration
     ansible.builtin.file:
       path: "{{ condor_config_root }}/config.d/00-htcondor-9.0.config"
       state: absent
     notify:
     - Reload HTCondor
-  - name: Set HTCondor role on all hosts
-    ansible.builtin.copy:
-      dest: "{{ condor_config_root }}/config.d/{{ role_file }}"
-      mode: 0644
-      content: |
-        use role:{{ htcondor_role }}
-    notify:
-    - Reload HTCondor
-  - name: Set HTCondor Central Manager and trust domain on all hosts
-    ansible.builtin.copy:
-      dest: "{{ condor_config_root }}/config.d/{{ pool_file }}"
-      mode: 0644
-      content: |
-        CONDOR_HOST={{ htcondor_central_manager_ips }}
-        UID_DOMAIN={{ trust_domain }}
-        TRUST_DOMAIN={{ trust_domain }}
-    notify:
-    - Reload HTCondor
+  - name: Create Toolkit configuration file
+    register: config_update
+    changed_when: config_update.rc == 137
+    failed_when: config_update.rc != 0 and config_update.rc != 137
+    ansible.builtin.shell: |
+      set -e -o pipefail
+      REMOTE_HASH=$(gcloud --format="value(md5_hash)" storage hash {{ config_object }})
+
+      CONFIG_FILE="{{ condor_config_root }}/config.d/{{ ghpc_config_file }}"
+      if [ -f "${CONFIG_FILE}" ]; then
+          LOCAL_HASH=$(gcloud --format="value(md5_hash)" storage hash "${CONFIG_FILE}")
+      else
+          LOCAL_HASH="d41d8cd98f00b204e9800998ecf8427e"
```

Comment:
Why this? Add comment to explain magic hash.

Reply:
It's perhaps a bit too clever, but one has to define the hash of an empty string (null terminated, 0 length) and this is it. In principle, if we somehow rendered an empty config file and had no file in place at all then no new (empty) file would be generated. Same in principle, different in detail so I'll modify it to read something entirely invalid (e.g. upper-case characters) and "self documenting."